Sources/PackageDescription/Version.swift

```diff
@@ -96,11 +96,33 @@ public struct Version {
 
 extension Version: Comparable {
     // Although `Comparable` inherits from `Equatable`, it does not provide a new default implementation of `==`, but instead uses `Equatable`'s default synthesised implementation. The compiler-synthesised `==`` is composed of [member-wise comparisons](https://github.com/apple/swift-evolution/blob/main/proposals/0185-synthesize-equatable-hashable.md#implementation-details), which leads to a false `false` when 2 semantic versions differ by only their build metadata identifiers, contradicting SemVer 2.0.0's [comparison rules](https://semver.org/#spec-item-10).
+    
+    /// Returns a Boolean value indicating whether two values are equal.
+    ///
+    /// Equality is the inverse of inequality. For any values `a` and `b`, `a ==
+    /// b` implies that `a != b` is `false`.
+    ///
+    /// This is the default implementation of the equal-to operator (`==`) for
+    /// any type that conforms to `Equatable`.
+    ///
+    /// - Parameters:
+    ///   - lhs: A value to compare.
+    ///   - rhs: Another value to compare.
     @inlinable
     public static func == (lhs: Version, rhs: Version) -> Bool {
         !(lhs < rhs) && !(lhs > rhs)
     }
-
+    
+    /// Returns a Boolean value indicating whether the value of the first
+    /// argument is less than that of the second argument.
+    ///
+    /// This function is the only requirement of the `Comparable` protocol. The
+    /// remainder of the relational operator functions are implemented by the
+    /// standard library for any type that conforms to `Comparable`.
+    ///
+    /// - Parameters:
+    ///   - lhs: A value to compare.
+    ///   - rhs: Another value to compare.
```

Comment:
I think these are unnecessary, since the documentation should automatically inherit from `Equatable`'s and `Comparable`'s. However, it might be beneficial to replace the generic explanations of `==` and `<` with ones specific to semantic versions, with notes on things that users should watch for (e.g. build metadata identifiers, numeric vs. alphanumeric identifier precedence, pre-release precedence, etc).

Reply:
Good point, definitely for the `<` function, I agree, because it's completely adapted for the purpose. I'll describe that in more detail. The others are more generic. I'll think about the `Equatable` conformance documentation, but for now, I am inclined to leave as-is.

Reply:
Some of the description in `==`'s documentation doesn't make sense. For example, this line: "This is the default implementation of the equal-to operator (`==`) for any type that conforms to `Equatable`.", since `Version` is not using the default implementation of `==`.

In general, I suggest that it's better to write the documentation for each function (not just these 2, but all functions in this PR) from scratch, instead of using the protocol's default implementation's, for the following reason:

1. If the conforming type is _not_ using the default implementation, then the documentation for the default implementation likely is incorrect for the custom implementation.

2. If the conforming type _is_ using the default implementation, then there is really no need to reuse the same documentation, because the same documentation should be automatically inherited.